### PR TITLE
Add Docker CI & support for ARM64 builds

### PR DESCRIPTION
**Pull request by Half-Shot**
We're looking to start using this in arm environments, so it would be useful to start building ARM images.

### CODE REVIEW

File: .github/workflows/docker-hub-latest.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        if: ${{ env.PUSH == 'true' }}
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_TOKEN }}
```
Comment (DMRobertson) [inline .github/workflows/docker-hub-latest.yml:35]: You're probably going to have to ask SRE/IT to share the secret with this repo, if it isn't already

Comment (DMRobertson) [reply .github/workflows/docker-hub-latest.yml:35]: Doesn't look like it's shared, from https://github.com/matrix-org/matrixmon/settings/secrets/actions

You could also upload to ghcr.io if that's any easier.

Comment (Half-Shot) [reply .github/workflows/docker-hub-latest.yml:35]: Unsure whether to maintain the status quo or ghcr.io. Given how little this image is updated, we might as well go with ghcr.io and tell people to migrate?

Comment (DMRobertson) [reply .github/workflows/docker-hub-latest.yml:35]: Oh, I didn't realise there was an existing image on dockerhub for this. Would be nice to keep that updated if we can... but no strong opinions from me either way.

Comment (Half-Shot) [reply .github/workflows/docker-hub-latest.yml:35]: I've agonized over this one for a day now and I'm really unsure. ghcr.io is convenient from a configuration point of view but it's a breaking change....

I think I'm going to revert what I did and go back to dockerhub. Instead, I'll ask ops for some lovely tokens.

**Comment by Half-Shot** [comment]
The correct dockerhub tokens have been shared fwiw.

**Review by DMRobertson — COMMENTED** [review]
Seems reasonable to me. Has there been a trial run of this? I see https://github.com/matrix-org/matrixmon/actions/runs/6625595439/job/17996866220?pr=11 which seems to have take 2 hours to run!!!

**Review by DMRobertson — APPROVED** [review]

**Comment by Half-Shot** [comment]
matrixmon seems to need a *lot* of dependencies, and it's all going through the slow slow muck that is qemu to do it. It's a real pain, but the time investment basically isn't worth it for now I think.

**Pull request merged**